app/models/ability.rb

```diff
@@ -35,7 +35,42 @@ def initialize(user)
     #   end
 
     user ||= User.new # guest user (not logged in)
+    id = user.id
+
     # All users can only update their own user attributes
-    can :update, User, id: user.id
+    alias_action :create, :read, :update, :destroy, :to => :crud
+
+    can :crud, Team
+    can :update, User, id: id
+
+    can :assign_ownership, Team, Team do |team|
+      team.owners.include? user
+    end
+
+    can :delete_ownership, Team, Team do |team|
+      team.owners.include? user
+      team.owners.length > 1
+    end
+
+    can :delete_membership, Team, Team do |team|
+      team.owners.include? user and Ability.num_owners(team, team_member) > 0
+    end
+
+    can :delete_membership, Team, Team do |team|
+      Ability.num_owners(team, team_member) > 0 and Integer(id) == Integer(team_member)
+    end
   end
+
+  private
+
+  def self.num_owners(team, team_member)
```

Comment:
It seems like the indentation of this method has gone wrong … to my mind, it should look like:

```ruby
  private
    def self.num_owners(team, team_member)
      owners = team.owners
      another_user = User.find(team_member)
      if owners.include? another_user
        owners_after_delete = owners - [another_user]
      else
        owners_after_delete = owners
      end
      owners_after_delete.length
    end
```